x/mongo/driver/crypt.go

```diff
@@ -267,7 +267,7 @@ func (c *Crypt) decryptKey(ctx context.Context, kmsCtx *mongocrypt.KmsContext) e
 		addr = fmt.Sprintf("%s:%d", host, defaultKmsPort)
 	}
 
-	conn, err := tls.Dial("tcp", addr, &tls.Config{})
+	conn, err := tls.Dial("tcp", addr, &tls.Config{MinVersion: tls.VersionTLS12})
```

Comment:
This is an actual change in driver behavior (all other changes in this PR are expected to cause zero behavior changes). All TLS connections dialed to KMS providers (AWS KMS, Azure Key Vault, GCP Key Management, Hashicorp Vault, or any local KMS provider) now must support TLS v1.2+.

Reply:
Note, the local KMS provider does not create TLS connections.

CC @markbenvenuto in case this raises alarm bells. I think assuming AWS, Azure, GCP, and any KMIP server has TLS 1.2+ is a reasonable assumption.

Reply:
Yes, this is a reasonable assumption. AWS, Azure and GCP all support 1.2. I could imagine somebody having an old KMIP server but they should upgrade and I do not want to hold them back.